connect-inject/handler.go

```diff
@@ -215,238 +216,132 @@ type Handler struct {
-				Message: fmt.Sprintf("Error configuring injection init container: %s", err),
-			},
-		}
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error configuring injection init container: %s", err))
 	}
-	patches = append(patches, addContainer(
-		pod.Spec.InitContainers,
-		[]corev1.Container{initCopyContainer, initContainer},
-		"/spec/initContainers")...)
+	pod.Spec.InitContainers = append(pod.Spec.InitContainers, initContainer)
 
 	// Add the Envoy and Consul sidecars.
-	esContainer, err := h.envoySidecar(&pod, req.Namespace)
+	esContainer, err := h.envoySidecar(pod, req.Namespace)
 	if err != nil {
 		h.Log.Error("Error configuring injection sidecar container", "err", err, "Request Name", req.Name)
-		return &v1beta1.AdmissionResponse{
-			Result: &metav1.Status{
-				Message: fmt.Sprintf("Error configuring injection sidecar container: %s", err),
-			},
-		}
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error configuring injection sidecar container: %s", err))
 	}
-	connectContainer, err := h.consulSidecar(&pod)
+	pod.Spec.Containers = append(pod.Spec.Containers, esContainer)
+
+	connectContainer, err := h.consulSidecar(pod)
 	if err != nil {
 		h.Log.Error("Error configuring consul sidecar container", "err", err, "Request Name", req.Name)
-		return &v1beta1.AdmissionResponse{
-			Result: &metav1.Status{
-				Message: fmt.Sprintf("Error configuring consul sidecar container: %s", err),
-			},
-		}
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error configuring consul sidecar container: %s", err))
 	}
-	patches = append(patches, addContainer(
-		pod.Spec.Containers,
-		[]corev1.Container{esContainer, connectContainer},
-		"/spec/containers")...)
+	pod.Spec.Containers = append(pod.Spec.Containers, connectContainer)
 
-	// Add annotations so that we know we're injected
-	patches = append(patches, updateAnnotation(
-		pod.Annotations,
-		map[string]string{
-			annotationStatus: injected,
-		})...)
+	// The annotations here have already been initialized be h.defaultAnnotations()
+	// and do not need to be checked for being a nil value.
+	pod.Annotations[annotationStatus] = injected
 
-	// Add annotations for metrics
-	promAnnotations, err := h.prometheusAnnotations(&pod)
+	// Add annotations for metrics.
+	promAnnotations, err := h.prometheusAnnotations(pod)
 	if err != nil {
 		h.Log.Error("Error configuring prometheus annotations", "err", err, "Request Name", req.Name)
-		return &v1beta1.AdmissionResponse{
-			Result: &metav1.Status{
-				Message: fmt.Sprintf("Error configuring prometheus annotations: %s", err),
-			},
-		}
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error configuring prometheus annotations: %s", err))
 	}
-	if promAnnotations != nil {
-		patches = append(patches, updateAnnotation(
-			pod.Annotations,
-			promAnnotations)...)
+	for key, value := range promAnnotations {
+		pod.Annotations[key] = value
 	}
```

Comment:
Not a blocker, but this could potentially move to the `prometheusAnnotations` function. We can just let it modify the annotations on the pod similar to how we do it with `defaultAnnotations`.